### PR TITLE
Add .travis.yml to the project

### PR DESCRIPTION
**Pull request by DavidBronfen**
#3

### CODE REVIEW

**Comment by nirgn975** [comment]
Couple of tips:
1. Create an issue for every PR.
2. Assign the issue and the PR to yourself (I know it's feels funny at first, but it's a good habit).
3. Look at the error (read them, they usually say a lot!)

And if you need me, just ask stuff on an issue or PR and I'll get a notification..

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+before_install:
+  - export CHROME_BIN=chromium-browser
+  - export DISPLAY=:99.0
+  - sh -e /etc/init.d/xvfb start
+
+
+script:
+  # Client
+  - cd ../client
```
Comment (nirgn975) [inline .travis.yml]: No. As you can see by the error message, you don't have a `client` directory, in this path.
job__10_1_-_davidbronfen_retrogusto_-_travis_ci

If you'll go to my `traivs.yml` you'll see I first enter to the `server` directory and then I need to enter to the `client` directory, so I go back (`cd ..`) and enter to the client directory (`cd client`) in one command (`cd ../client`).

But you don't enter to your `server` directory, you just want to enter to your `client` directory in the first place, so you don't have to go back..

If you're not sure where you're in travis, do a commit with the commands `pwd` and `ls` so you'll see the path and all files/directories in this path.
Example:
```
script:
  - pwd
  - ls
  - cd ../client
...
...
```

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+
+sudo: required
+dist: trusty
+language: node_js
+node_js:
+  - "6"
+  - "7"
+
+addons:
+apt:
```
Comment (nirgn975) [inline .travis.yml]: You're missing an indentation for this block (see [here](https://github.com/nirgn975/Expenses/blob/master/.travis.yml#L4-L9))

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+
```
Comment (nirgn975) [inline .travis.yml]: There is no need for this whitespace, delete it.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  packages:
+    - google-chrome-stable
+    - google-chrome-beta
+
+cache:
+  npm: true
+  directories:
+  - client/node_modules
+    
+
```
Comment (nirgn975) [inline .travis.yml]: One new line is enough, stay consistent.

**Comment by nirgn975** [comment]
I was stack on this for quite some time myself. The `npm test` command wait for changes. You can edit this be add a flag to the `npm test` command.

As you can see in the `package.json` file, `npm test` command does `ng test`. If you'll add a some flag there that told the `ng test` command to not wait for changes, then the `npm test` will work as expected.

Read the section `Running unit tests` [here](https://github.com/angular/angular-cli/blob/master/docs/documentation/overview.md)

**Comment by DavidBronfen** [comment]
Trying to test only the lint, to find which one is failing the test.

**Comment by nirgn975** [comment]
Try remove the line with `- google-chrome-beta`

**Comment by DavidBronfen** [comment]
Removing all the sections:
```
cache:
  npm: true
  directories:
  - client/node_modules

before_install:
  - export CHROME_BIN=chromium-browser
  - export DISPLAY=:99.0
  - sh -e /etc/init.d/xvfb start

install:
  - cd client && npm install

script:
  # Client
  - npm run lint
  - npm run test
  - npm run e2e
  - ng build --prod -- aot
```
To find who to blame.

**Comment by nirgn975** [comment]
I think it's something with the `addons`

**Comment by DavidBronfen** [comment]
@nirgn975 Maybe it's just a bad day to configure Travis?

**Comment by DavidBronfen** [comment]
@nirgn975 Do i need to add a `package.json` file as they say [here](http://karma-runner.github.io/0.8/plus/Travis-CI.html) in the 'Setup a Test Command' section?

**Comment by DavidBronfen** [comment]
@nirgn975 I added also the `customLaunchers` to the `karma.conf.js` as I've read [here](http://blog.500tech.com/setting-up-travis-ci-to-run-tests-on-latest-google-chrome-version/).

But it's still fails:
image

File: npm-debug.log [hunk 5]
```diff
@@ -0,0 +1,20 @@
+0 info it worked if it ends with ok
```
Comment (nirgn975) [inline npm-debug.log]: Remove this file, this should be in `.gitignore`

File: client/karma.conf.js [hunk 6]
```diff
@@ -36,6 +36,14 @@ module.exports = function (config) {
     logLevel: config.LOG_INFO,
     autoWatch: true,
     browsers: ['Chrome'],
-    singleRun: false
+    singleRun: false,
```
Comment (nirgn975) [inline client/karma.conf.js]: I don't think you need those changes, I know I don't have them.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+cache:
+  npm: true
+  directories:
+  - client/node_modules
+
+install:
+  - cd client && npm install
+
+before_script:
+  - export DISPLAY=:99.0
```
Comment (nirgn975) [inline .travis.yml]: This should your `before_script`
```
  - export CHROME_BIN=chromium-browser
  - export DISPLAY=:99.0
  - sh -e /etc/init.d/xvfb start
```

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
+sudo: required
+dist: trusty
+language: node_js
+node_js:
+  - "7"
+
+before_install:
+ - export CHROME_BIN=/usr/bin/google-chrome
```
Comment (nirgn975) [inline .travis.yml]: Try to remove this `before_install` commands, Travis provide chrome as an addon, I don't think you should install it by yourself (for now).

File: .gitignore [hunk 9]
```diff
@@ -0,0 +1 @@
+_.travis.yml
```
Comment (nirgn975) [inline .gitignore]: Why?

**Comment by nirgn975** [comment]
Also, edit your `package.json` file. Replace your `test` command with:
```
"test": "ng test --single-run --code-coverage",
```

File: .gitignore (hunk 9 above)
Comment (DavidBronfen) [inline .gitignore]: Just added a temporary `yml` to remember the commands has been removed.

File: npm-debug.log (hunk 5 above)
Comment (nirgn975) [inline npm-debug.log]: Don't forget to delete this file, and commit after it.

File: .travis.yml [hunk 10]
```diff
@@ -0,0 +1,31 @@
+sudo: required
+dist: trusty
+language: node_js
+node_js:
+  - "7"
+addons:
```
Comment (nirgn975) [inline .travis.yml]: Add a new line before `addons`

**Comment by nirgn975** [comment]
when you ready for review, mention me

**Comment by DavidBronfen** [comment]
@nirgn975 I think i'm done here. 
It's still shows a changes requested message. is it a bug or something? 
I can't find the review that i need to modify.

**Comment by nirgn975** [comment]
Looking good. Last thing, showoff and add the badge to your `README.md` file:

Right after the `# Retrogusto`:
```
[![Build Status][travis-image]][travis-url]
```

At the end of the file:
```
[travis-image]: https://travis-ci.org/DavidBronfen/Retrogusto.svg?branch=master
[travis-url]: https://travis-ci.org/DavidBronfen/Retrogusto
```

**Comment by DavidBronfen** [comment]
@nirgn975 Ready to final review.

**Pull request merged**